File: exercises/imperative-to-functional/03/example-solution.rb
require 'json'

rid, = ARGV

data = JSON.parse(STDIN.read)

data['exams'].select { |exam| exam['student_id'] == rid }
             .map { |exam| exam['timeslot'] }
             .tally
             .select { |timeslot, frequency| frequency > 1 }
             .map(&:first)
             .sort
             .then { |result| puts result }

File: exercises/imperative-to-functional/03/example-solution.py
import collections
import json
import sys


data = json.loads(sys.stdin.read())
rid = sys.argv[1]

frequencies = collections.Counter( exam['timeslot'] for exam in data['exams'] if exam['student_id'] == rid )
result = sorted(timeslot for timeslot, frequency in frequencies.items() if frequency > 1)

print("\n".join(str(x) for x in result))
